CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -150,28 +146,51 @@ public class Child: NodeChoiceConvertible {
   /// For any other kind of child nodes, accessing this property crashes.
   public var syntaxChoicesType: TypeSyntax {
     precondition(kind.isNodeChoices, "Cannot get `syntaxChoicesType` for node that doesn’t have nodeChoices")
-    return "\(raw: name.withFirstCharacterUppercased)"
+    return "\(raw: newestName.withFirstCharacterUppercased)"
   }
 
   /// If this child only has tokens, the type that the generated `TokenSpecSet` should get.
   ///
   /// For any other kind of child nodes, accessing this property crashes.
   public var tokenSpecSetType: TypeSyntax {
     precondition(kind.isToken, "Cannot get `tokenSpecSetType` for node that isn’t a token")
-    return "\(raw: name.withFirstCharacterUppercased)Options"
+    return "\(raw: newestName.withFirstCharacterUppercased)Options"
   }
 
-  /// The deprecated name of this child that's suitable to be used for variable or enum case names.
-  public var deprecatedVarName: TokenSyntax? {
-    guard let deprecatedName = deprecatedName else {
-      return nil
+  /// If this child is actually part of another child's history, links back
+  /// to that child (and its parent, if this child was extracted to another
+  /// node). Empty if this is the newest version of the child.
+  public let newerChildPath: [Child]
+
+  /// True if this child was created by a `Refactoring`. Such children are part
+  /// of a compatibility layer and are therefore deprecated.
+  public var isHistorical: Bool {
+    !newerChildPath.isEmpty
+  }
+
+  /// Replaces the nodes in `newerChildPath` with their own `newerChildPath`s,
+  /// if any, to form a child path enitrely of non-historical nodes.
+  public func makeNewestChildPath() -> [Child] {
```

Comment:
Should either `newerChildPath` or `makeNewestChildPath()` be `private`. It seems like it’s non-trivial for clients to decide whether they want `makeNewestChildPath()` or `newerChildPath`. Or do we need both in different circumstances?

Reply:
We may be able to get away with computing and storing the `newestChildPath` at initialization time. I'll consider that.